### DISCUSSION TITLE
Provide a way to set custom cache directory

### DISCUSSION THREAD

**Discussion by kirill-konshin** [vercel/next.js#72528] [Ideas]
### Link to the code that reproduces this issue

https://github.com/kirill-konshin/next-electron-rsc/tree/main/demo

### To Reproduce

1. Clone the repo
2. `cd demo`
3. `yarn build`
4. `yarn open`

### Current vs. Expected behavior

Next.js is in standalone mode here, and is bundled using `electron-builder`.

Since the `.next/standalone/demo/` is part of `files`, it becomes part of the ASAR.

When the Electron runs as a build, Next.js emits following error:

```
 ⨯ Failed to write image to cache 24-d7GMv9Put-b8LiQHcADsQ3iL099cDsuFLFsgj0LE Error: ENOTDIR: not a directory, mkdir '/x/demo/dist/mac-arm64/Next Electron RSC.app/Contents/Resources/app.asar/.next/standalone/demo/.next/cache/images/24-d7GMv9Put-b8LiQHcADsQ3iL099cDsuFLFsgj0LE'
    at async writeToCacheDir (/x/demo/dist/mac-arm64/Next Electron RSC.app/Contents/Resources/app.asar/.next/standalone/demo/node_modules/next/dist/server/image-optimizer.js:213:5)
    at async ImageOptimizerCache.set (/x/demo/dist/mac-arm64/Next Electron RSC.app/Contents/Resources/app.asar/.next/standalone/demo/node_modules/next/dist/server/image-optimizer.js:507:13)
    at async (/x/demo/dist/mac-arm64/Next Electron RSC.app/Contents/Resources/app.asar/.next/standalone/demo/node_modules/next/dist/server/response-cache/index.js:128:25)
    at async (/x/demo/dist/mac-arm64/Next Electron RSC.app/Contents/Resources/app.asar/.next/standalone/demo/node_modules/next/dist/lib/batcher.js:45:32) {
  errno: -20,
  code: 'ENOTDIR',
  syscall: 'mkdir',
  path: '/x/demo/dist/mac-arm64/Next Electron RSC.app/Contents/Resources/app.asar/.next/standalone/demo/.next/cache/images/24-d7GMv9Put-b8LiQHcADsQ3iL099cDsuFLFsgj0LE'
}
```

So either ASAR is not writable at all (https://www.electronjs.org/docs/latest/tutorial/asar-archives#archives-are-read-only) the solution could be to provide alternative `cacheDir` outside of ASAR: https://github.com/vercel/next.js/blob/canary/packages/next/src/server/image-optimizer.ts#L372

I tried opting cache path out of ASAR, did not work for a different reason: https://github.com/electron-userland/electron-builder/issues/7264

Original issue: https://github.com/kirill-konshin/next-electron-rsc/issues/3

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.0.0: Tue Sep 24 23:39:07 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T6000
  Available memory (MB): 65536
  Available CPU cores: 10
Binaries:
  Node: 23.0.0
  npm: 10.9.0
  Yarn: 4.5.1
  pnpm: N/A
Relevant Packages:
  next: 15.0.2
  eslint-config-next: 15.0.2
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.6.3
Next.js Config:
  output: standalone
```

### Which area(s) are affected? (Select all that apply)

Image (next/image)

### Which stage(s) are affected? (Select all that apply)

Other (Deployed)

**Comment by samcx** [comment]
@kirill-konshin Have you tried taking a look at the `cacheHandler` config → https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath?

**Reply by kirill-konshin** [reply]
Not sure if this one apply because it caches in memory, not the disk cache for images.

https://github.com/vercel/next.js/blob/canary/packages/next/src/server/image-optimizer.ts#L378 has no mentions of cache handler, and has fixed path for cache.

**Reply by ragingwind** [reply]
@samcx any updates for this?

**Reply by samcx** [reply]
@ragingwind No updates on this yet unfortunately!
